### ISSUE TITLE
Ask to create default policy for new qrexec services?

### ISSUE THREAD

**Issue by rootkovska** [QubesOS/qubes-issues#3005]
In 3.2 if the user creates a new qrexec service (for which there is no policy file in dom0), we get a window asking to create the default policy for the service. In 4.0-rc1 we get connection refusal if the policy is missing.

**Comment by marmarek** [comment]
Since now we have new qrexec prompt window - which is harder to accidentally accept, what do you think about applying default policy "$anyvm $anyvm ask", without asking first?

**Comment by rootkovska** [comment]
+1

**Comment by marmarek** [comment]
On the other hand, this would allow VM existence confirmation, which may be privacy issue. So, I'll implement explicit confirmation, as it was in 3.2.

**Comment by rootkovska** [comment]
Why would it allow VM existence confirmation any better?

**Comment by marmarek** [comment]
Because when target domain specified by source domain does not exist, the call will be rejected (not existing domain does not match with `$anyvm`, so `ask` action will not be applied).

**Comment by rootkovska** [comment]
But... here we're discussing non-existing services?

**Comment by marmarek** [comment]
Yes. And when you automatically get `$anyvm $anyvm ask` policy, it allows exactly what I've described above. That's why I prefer to ask before policy creation (before even looking at target domain).

**Comment by rootkovska** [comment]
Is there anything fundamental that prevents us from not dropping the call before we display the window? Or perhaps adding a random delay in case non-existing target?

**Comment by rootkovska** [comment]
(Random delay simulating the user reading and click on the dialog)

**Comment by marmarek** [comment]
> Is there anything fundamental that prevents us from not dropping the call before we display the window?

Can you rephrase? We _are_ actually dropping the call before displaying the window.

Note that the "problem" is not "ask" action rejecting non-existing target, but not applying "ask" action at all (so it gets to implicit deny, because no rule have matched).

As for random delay - I think it may break many legitimate use cases... And getting it right will be really hard - for example we need to assume some range, and it should match how much _this particular user_ spends on reading the dialog.

IMO one time ask before policy creation, at the first service call is much simpler solution - both in terms of implementation and usability.

**Comment by marmarek** [comment]
> Can you rephrase? We are actually dropping the call before displaying the window.

Ah, there is "not". Ok, so ignore this and read the rest of my comment.
